app/controllers/devise/two_step_verification_controller.rb

```diff
@@ -1,10 +1,18 @@
 class Devise::TwoStepVerificationController < DeviseController
-  before_filter :prepare_and_validate
+  before_filter :prepare_and_validate, except: [:prompt, :defer]
   skip_before_filter :handle_two_step_verification
 
   attr_reader :otp_secret_key
   private :otp_secret_key
 
+  def prompt
+  end
+
+  def defer
+    current_user.defer_two_step_verification
+    redirect_to root_path, notice: 'You will not be required to setup 2-step verification.'
```

Comment:
Two things here:
* this should redirect them to the place they were going to before they were asked to sign in (eg another app, or a specific page in signon). The root_path would only be used as a fallback. I think Devise handles this for a normal signin?
* chatting with @fofr he thinks we can leave the `notice` out